master/internal/api_trials.go

```diff
@@ -645,6 +645,40 @@ func (a *apiServer) GetExperimentTrials(
 	return resp, nil
 }
 
+func (a *apiServer) GetTrialRemainingLogRetentionDays(
+	ctx context.Context, req *apiv1.GetTrialRemainingLogRetentionDaysRequest,
+) (resp *apiv1.GetTrialRemainingLogRetentionDaysResponse, err error) {
+	t, err := db.TrialByID(ctx, int(req.Id))
+	if err != nil {
+		return nil, fmt.Errorf("getting trial %v: %w", req.Id, err)
+	}
+
+	q := `
+SELECT   
+CASE
+	WHEN MIN(t.end_time) <= ( retention_timestamp() - make_interval(days => ?) ) THEN 0
```

Comment:
qq: whats the difference between `transaction_timestamp()` and `retention_timestamp()`

Reply:
From what I can tell, only that `retention_timestamp()` isn't a Postgres built-in (it just usually calls `transaction_timestamp()`, which allows tests to redefine it instead of having to mess with the clock.

Reply:
found it https://github.com/determined-ai/determined/blob/d57febad88b174de01dc0ede4b35ce26dc7d28c4/master/static/migrations/20240326162830_log-retention.tx.up.sql#L4